modules/@apostrophecms/attachment/index.js

```diff
@@ -1037,6 +1042,66 @@ module.exports = {
           });
         }
         next();
+      },
+      // This migration is needed because formerly,
+      // docs that only referenced this attachment via
+      // a join were counted as "owning" it, which is
+      // incorrect and leads to failure to make it
+      // unavailable at the proper time. The name was
+      // changed to ensure this migration would run
+      // again after that bug was discovered and fixed.
+      addDocReferencesContainedMigration() {
+        self.apos.migration.add(self.__meta.name + '.docReferencesContained', self.recomputeAllDocReferences);
+      },
+      // Recompute the `docIds` and `archivedDocIds` arrays
+      // from scratch. Should only be needed by the
+      // one-time migration that fixes these for older
+      // databases, but can be run at any time via the
+      // `apostrophe-attachments:recompute-doc-references`
+      // task, just in case the need arises or your site
+      // was affected by the very brief availability of 2.77.0
+      // which effectively marked all attachments as
+      // not in use.
+      async recomputeAllDocReferences() {
+        const attachmentUpdates = {};
+        await self.apos.migration.eachDoc({}, 5, addAttachmentUpdates);
```

Comment:
This works, but I'm confused. They're in aposAttachments, but my understanding is that eachDoc only hits aposDocs.

Reply:
Correct, but we use what we learn from scanning those documents to build an array of mongodb bulk operations that are then executed against aposAttachments (self.db, in the attachment module).